### PR TITLE
feat: add yamlfix

### PR DESCRIPTION
**Pull request by Psycojoker**
Add yamlfix to uniformise yaml formatting everywhere.

Related ClickUp, GitHub or Jira tickets : ALEPH-180

## Self proofreading checklist

- [x] New code does not duplicate the functions of builtin or popular libraries.

## How to test

Just run:

```
hatch run lint:style
hatch run lint:fmt
```

### CODE REVIEW

File: .github/workflows/build-deb-package.yml [hunk 1]
```diff
@@ -1,90 +1,78 @@
-name: "Build Packages"
-on:
-  push
-
+---
+name: Build Packages
+on: push
 jobs:
   build_deb:
-    name: "Build ${{ matrix.os }} Package"
```
Comment (olethanh) [inline .github/workflows/build-deb-package.yml:7]: I honestly don't think we should encourage string in yaml without quote, is too error prone.

Comment (Psycojoker) [reply .github/workflows/build-deb-package.yml:7]: We've decided to add `preserve_quotes = true` in yamlfix config to avoid this issue

**Comment by hoh** [comment]
I find the result less readable without the empty lines. What is your opinion on this ?

**Comment by Psycojoker** [comment]
> I find the result less readable without the empty lines. What is your opinion on this ?

It makes sens to keep at least one whiteline when there is one yes, let's do that https://lyz-code.github.io/yamlfix/#whitelines-adjusting

image

**Comment by Psycojoker** [comment]
Done, I've also added `section_whitelines = 2` to match python behavior

**Review by olethanh — APPROVED** [review]

**Pull request merged**